app/kumactl/cmd/generate/generate_certificate_test.go

```diff
@@ -0,0 +1,74 @@
+package generate_test
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/Kong/kuma/app/kumactl/cmd"
+	"github.com/Kong/kuma/app/kumactl/cmd/generate"
+	"github.com/Kong/kuma/pkg/tls"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"io/ioutil"
+)
+
+var _ = Describe("kumactl generate certificate", func() {
+
+	var backupNewSelfSignedCert func(string, tls.CertType, ...string) (tls.KeyPair, error)
+	BeforeEach(func() {
+		backupNewSelfSignedCert = generate.NewSelfSignedCert
+	})
+	AfterEach(func() {
+		generate.NewSelfSignedCert = backupNewSelfSignedCert
+	})
+
+	BeforeEach(func() {
+		generate.NewSelfSignedCert = func(string, tls.CertType, ...string) (tls.KeyPair, error) {
+			return tls.KeyPair{
+				CertPEM: []byte("CERT"),
+				KeyPEM:  []byte("KEY"),
+			}, nil
+		}
+	})
+
+	It("should generate certificate", func() {
```

Comment:
Why don't we have a test case for server cert (to make sure that cert type `server` is passed into `NewSelfSignedCert ()` in that case)  ?